src/plugins/intel_cpu/src/nodes/dft.cpp

```diff
@@ -404,24 +481,221 @@ void DFT::naiveDFT(float* data, size_t dataLength) const {
+            fftKernel.reset(new jit_uni_fft_kernel_f32<cpu::x64::avx2>(jdp));
+        } else if (mayiuse(cpu::x64::sse41)) {
+            fftKernel.reset(new jit_uni_fft_kernel_f32<cpu::x64::sse41>(jdp));
+        } else {
+            IE_THROW() << "Can't create jit FFT kernel";
+        }
+
+        if (fftKernel)
+            fftKernel->create_ker();
+    }
+}
+
+float* DFT::DFTJitExecutor::fft(float* inBuffer,
+                                float* outBuffer,
+                                int64_t dataLength,
+                                bool inverse,
+                                bool parallelize) const {
+    static int cacheSizeL3 = dnnl::utils::get_cache_size(3, false);
+    static int elementsPerCacheLine = cacheSizeL3 / sizeof(float);
+    size_t nComplex = dataLength / 2;
+
+    auto blockIteration =
+        [&](const size_t block, const size_t numBlocks, const size_t blockSize, const size_t nextIterationBlockSize) {
+            auto arg = jit_args_fft();
+
+            arg.src = inBuffer + block * nextIterationBlockSize * 2;
+            arg.dst = outBuffer + block * nextIterationBlockSize;
+            arg.twiddles = &twiddlesFFT[(numBlocks + block - 1) * 2];
+            arg.num_blocks = numBlocks;
+            arg.work_amount = nextIterationBlockSize;
+            arg.n_complex = nComplex;
+
+            (*fftKernel)(&arg);
+        };
+
+    size_t blockSize;
+    size_t nextIterationBlockSize = dataLength;
+    for (size_t numBlocks = 1; numBlocks < nComplex; numBlocks *= 2) {
+        blockSize = nextIterationBlockSize;
+        nextIterationBlockSize /= 2;
+        if (parallelize && blockSize >= 4 * elementsPerCacheLine) {
+            parallel_for(numBlocks, [&](const size_t block) {
+                blockIteration(block, 1, blockSize, nextIterationBlockSize);
+            });
+        } else {
+            blockIteration(0, numBlocks, blockSize, nextIterationBlockSize);
+        }
+
+        std::swap(inBuffer, outBuffer);
+    }
+
+    if (inverse) {
+        for (int64_t k = 0; k < dataLength; k++) {
+            inBuffer[k] /= nComplex;
+        }
+    }
+
+    return inBuffer;
+}
+
+void DFT::DFTJitExecutor::naiveDFT(float* data, size_t dataLength, bool inverse) const {
+    std::vector<float> outputBuffer(dataLength);
+    const size_t nComplex = dataLength / 2;
+    const auto& twiddles = twiddlesMapDFT.find(nComplex)->second;
+
+    parallel_for(nComplex, [&](size_t k) {
+        auto arg = jit_args_dft();
+
+        arg.src = data;
+        arg.dst = outputBuffer.data() + 2 * k;
+        arg.twiddles = twiddles.data() + 2 * k * nComplex;
+        arg.work_amount = nComplex;
+        arg.index = k;
+
+        (*dftKernel)(&arg);
+    });
+    cpu_memcpy(data, outputBuffer.data(), dataLength * sizeof(float));
+}
+
 bool DFT::created() const {
     return getType() == Type::DFT;
 }
 
-void DFT::createPrimitive() {}
+bool DFT::needPrepareParams() const {
+    return lastInverse != inverse;
```

Comment:
That is not quite right behavior. The *needPrepareParams()* function should check whether the parameters related to shape have been changed and return true if need to call *prepareParams()* in this case. So default behavior is enough here.

Reply:
I restored the original behaviour with additional check - kernel need to be recompiled when 'inverse' flag changes